sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,125 @@
-import {
-  AppBar,
-  AppBarProps,
-  Backdrop,
-  IconButton,
-  paperClasses,
-  Popover,
-  PopoverProps,
-  Typography
-} from '@mui/material';
-import CloseRounded from '@mui/icons-material/CloseRounded';
+import { AppBarProps, paperClasses, Popover, PopoverProps, useTheme } from '@mui/material';
 import ChatGPT, { ChatGPTProps } from './ChatGPT.tsx';
-import React from 'react';
+import React, { useState } from 'react';
+import MinimizedBar from '@craftercms/studio-ui/components/MinimizedBar';
+import DialogHeader from '@craftercms/studio-ui/components/DialogHeader/DialogHeader';
+import AlertDialog from '@craftercms/studio-ui/components/AlertDialog';
+import PrimaryButton from '@craftercms/studio-ui/components/PrimaryButton/PrimaryButton';
+import SecondaryButton from '@craftercms/studio-ui/components/SecondaryButton/SecondaryButton';
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
+  const theme = useTheme();
   const {
     open,
     onClose,
-    appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const [openAlertDialog, setOpenAlertDialog] = useState(false);
+
+  const handleMinimize = () => {
+    onMinimize?.();
+  };
+
   return (
-    <Popover
-      open={open}
-      onClose={onClose}
-      keepMounted={false}
-      anchorReference="none"
-      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-      anchorPosition={{ top: 100, left: 100 }}
-      BackdropProps={{ invisible: false, sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' } }}
-      {...popoverProps}
-      sx={{
-        [`> .${paperClasses.root}`]: {
-          width,
-          height,
-          display: 'flex',
-          flexDirection: 'column',
-          bottom: 10,
-          right: 10
-        },
-        ...popoverProps?.sx
-      }}
-    >
-      <AppBar
-        position="static"
-        {...appBarProps}
+    <>
+      <Popover
+        open={open && !isMinimized}
+        onClose={() => setOpenAlertDialog(true)}
```

Comment:
Consider as a future improvement a way of only showing the warning if there is a conversation to be lost.

Reply:
This mechanism of dialogs is likely going away in favour of one that's more dynamic and supports multiple dialogs of the same type opened at the same time. 😬 

Also this makes the plugin a too coupled to the edit dialog. I can't remember what the state looks like when there's no dialog, but if nullish, this could even throw.

Overall, let's remove this.

My point was that currently, if you open and immediately close the chat without any conversation, the chat will still ask for confirmation when there's no conversation to be lost. It'd be good to have a way to inform the Chat popover container that there is a conversation so it doesn't prompt you for confirmation if there's is nothing to lose from closing.

Reply:
@rart Is this about checking if there is no conversation, we should just close the chat? If yes, please see the commit: https://github.com/craftercms/chatgpt-plugin/pull/4/commits/fd482b11e087e6e0fe04e6f827cd55ba0570eeef

I'm not quite sure about the following:
> Also this makes the plugin a too coupled to the edit dialog.
We show the warning in various cases, including open the edit dialog or other type of dialogs.